### PR TITLE
xds: create CommonXdsClient from ClientXdsClient and ServerXdsClient:

### PR DESCRIPTION
**Pull request by sanjaypujare**
part1 of the effort. Next part will include actually using CommonXdsClient on the server side

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/CommonXdsClient.java [hunk 1]
```diff
@@ -89,7 +90,7 @@
 /**
  * XdsClient implementation for client side usages.
  */
-final class ClientXdsClient extends AbstractXdsClient {
+final class CommonXdsClient extends AbstractXdsClient {
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/CommonXdsClient.java:93]: Why name it common? There will be just one kind of XdsClient.

File: xds/src/main/java/io/grpc/xds/CommonXdsClient.java [hunk 2]
```diff
@@ -1118,7 +1165,17 @@ public void run() {
...
+  void cancelLdsResourceWatch(String resourceName, LdsResourceWatcher watcher) {
+    commonLdsResourceWatchCancel(resourceName, watcher);
+  }
+
+  @Override
+  void cancelListenerResourceWatch(String resourceName, ListenerWatcher watcher) {
+    commonLdsResourceWatchCancel(resourceName, watcher);
+  }
+
+  private void commonLdsResourceWatchCancel(final String resourceName,
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/CommonXdsClient.java:1177]: This naming is not Java's convention (verb + noun).

File: xds/src/main/java/io/grpc/xds/CommonXdsClient.java [hunk 3]
```diff
@@ -244,12 +250,30 @@ protected void handleLdsResponse(String versionInfo, List<Any> resources, String
       }
       ldsUpdates.put(listenerName, update);
     }
+
+    // code to process serverSideListeners to generate ListenerUpdate’s
+    Map<String, ListenerUpdate> listenerUpdates = new HashMap<>();
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/CommonXdsClient.java:255]: We should first eliminate one of the Listener/LDS watchers before trying to put server side logic into this XdsClient implementation.

File: xds/src/main/java/io/grpc/xds/CommonXdsClient.java [hunk 4]
```diff
@@ -262,6 +286,20 @@ protected void handleLdsResponse(String versionInfo, List<Any> resources, String
     }
   }
 
+  private StructOrError<EnvoyServerProtoData.Listener> parseListener(Listener listener) {
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/CommonXdsClient.java:289]: Also, please clean up EnvoyServerProtoData first.

**Comment by sanjaypujare** [comment]
closing in favor of another one

**Pull request closed without merging**